### DISCUSSION TITLE
Notebook selection

### DISCUSSION THREAD

**Discussion by megalithic** [megalithic/zk.nvim#9] [Ideas]
Really wanting to get thoughts around notebook selection; for future note creation/linking/etc. Do we want to just support a single notebook to start with to simplify the whole thing (e.g., it only works for the current notebook you're located in via your `cwd`)? 

I originally was thinking we could utilize an easy tab completion mechanism to query the notebooks via `zk` to pipe it to FZF in neovim. Selecting your found notebook would set a plugin config var that would have all notes created under that autochdir'd `cwd`, and subsequent configured notebook for that path.

/cc @evantravers @mickael-menu

**Comment by mickael-menu** [comment]
> Do we want to just support a single notebook to start with to simplify the whole thing (e.g., it only works for the current notebook you're located in via your cwd)?

That could be best for the start, and more in line with how `zk` works in the shell. Until we add a global default notebook, if we do, a possible approach would be to have a shortcut to an "index" page, which would automatically put the current buffer in the notebook path. A bit like VimWiki's `<leader>ww` which opens the wiki index.

> I originally was thinking we could utilize an easy tab completion mechanism to query the notebooks via zk to pipe it to FZF in neovim. 

I'm afraid that would complicate things. But at the same time I use a single notebook, maybe that could be relevant in other situations...

**Comment by mickael-menu** [comment]
@megalithic I think that this feature would solve the problem: https://github.com/mickael-menu/zk/issues/13

Then `zk.nvim` doesn't need to do anything. If NeoVim cwd is in a notebook, it will be used by `zk`. Otherwise, if there's a global `ZK_NOTEBOOK_DIR` set, `zk` will use this instead.
